src/configuration/Preprocessor.h

```diff
@@ -59,6 +59,26 @@ operator <<(std::ostream &os, const Location &l)
 class PreprocessedDirective
 {
 public:
+    /// represents possible differences between two directives
+    class Diff {
+    public:
+        /// difference type bitmasks
+        enum class Scope: uint8_t {
```

Comment:
Can we go back to `unsigned int` here? `int` is arguably simpler/safer and might even be faster. I see nothing important we are gaining by insisting on using an (unsigned) char here. I did not find the answer in the commit that changed this type.

```suggestion
        enum class Scope: unsigned int {
```

Reply:
Ok, let's go back to int. I used uint8_t because this type should be enough to take in all possible (or future) values of the enum, and we do not need a bigger int.